From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Doe <john.doe@somewhere.on.planet>
Date: Tue, 20 Feb 2024 04:33:41 +0800
Subject: Patching kernel meson files
 arch/arm/boot/dts/amlogic/meson8b-onecloud.dts
 arch/arm/boot/dts/amlogic/meson8b.dtsi

Signed-off-by: John Doe <john.doe@somewhere.on.planet>
---
 arch/arm/boot/dts/amlogic/meson8b-onecloud.dts | 38 ++++++++++
 arch/arm/boot/dts/amlogic/meson8b.dtsi         | 17 +++++
 2 files changed, 55 insertions(+)

diff --git a/arch/arm/boot/dts/amlogic/meson8b-onecloud.dts b/arch/arm/boot/dts/amlogic/meson8b-onecloud.dts
index 6ed195220ea1..35a09582f02d 100644
--- a/arch/arm/boot/dts/amlogic/meson8b-onecloud.dts
+++ b/arch/arm/boot/dts/amlogic/meson8b-onecloud.dts
@@ -140,11 +140,23 @@ vcc_5v: regulator-vcc-5v {
 		vin-supply = <&p12v>;
 
 		regulator-boot-on;
 		regulator-always-on;
 	};
+	sdio_pwrseq: sdio-pwrseq {
+		compatible = "mmc-pwrseq-simple";
 
+		pinctrl-0 = <&xtal_32k_out_pins>;
+		pinctrl-names = "default";
+
+		reset-gpios = <&gpio GPIOX_11 GPIO_ACTIVE_LOW>,
+			      <&gpio_ao GPIOAO_6 GPIO_ACTIVE_LOW>;
+
+		clocks = <&xtal_32k_out>;
+		clock-names = "ext_clock";
+	};
+	
 	vcc_3v3: regulator-vcc-3v3 {
 		compatible = "regulator-fixed";
 
 		regulator-name = "VCC3V3";
 		regulator-min-microvolt = <3300000>;
@@ -153,10 +165,17 @@ vcc_3v3: regulator-vcc-3v3 {
 		vin-supply = <&p12v>;
 
 		regulator-boot-on;
 		regulator-always-on;
 	};
+	
+	xtal_32k_out: xtal-32k-out-clk {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <32768>;
+		clock-output-names = "xtal_32k_out";
+	};
 
 	vcc_1v8: regulator-vcc-1v8 {
 		compatible = "regulator-fixed";
 
 		regulator-name = "VCC1V8";
@@ -374,11 +393,30 @@ &sdhc {
 
 	mmc-pwrseq = <&emmc_pwrseq>;
 	vmmc-supply = <&vcc_3v3>;
 	// vqmmc-supply = <&vcc_3v3>;
 };
+// SDIO wifi
+&sdhc {
+	status = "okay";
+
+	pinctrl-0 = <&sdxc_a_pins>;
+	pinctrl-names = "default";
 
+	bus-width = <4>;
+	max-frequency = <50000000>;
+
+	disable-wp;
+	non-removable;
+	cap-mmc-highspeed;
+	cap-sd-highspeed;
+
+	mmc-pwrseq = <&sdio_pwrseq>;
+
+	vmmc-supply = <&vcc_3v3>;
+	vqmmc-supply = <&vcc_3v3>;
+};
 &sdio {
 	status = "okay";
 
 	pinctrl-0 = <&sd_b_pins>;
 	pinctrl-names = "default";
diff --git a/arch/arm/boot/dts/amlogic/meson8b.dtsi b/arch/arm/boot/dts/amlogic/meson8b.dtsi
index da9216eee98b..ae622d5df735 100644
--- a/arch/arm/boot/dts/amlogic/meson8b.dtsi
+++ b/arch/arm/boot/dts/amlogic/meson8b.dtsi
@@ -629,10 +629,19 @@ mux {
 				function = "sd_b";
 				bias-disable;
 			};
 		};
 
+		sdxc_a_pins: sdxc-a {
+			mux {
+				groups = "sdxc_d0_a", "sdxc_d13_a",
+					 "sdxc_d47_a", "sdxc_clk_a",
+					 "sdxc_cmd_a";
+				function = "sdxc_a";
+				bias-pull-up;
+			};
+		};
 		sdxc_c_pins: sdxc-c {
 			mux {
 				groups = "sdxc_d0_c", "sdxc_d13_c",
 					 "sdxc_d47_c", "sdxc_clk_c",
 					 "sdxc_cmd_c";
@@ -671,10 +680,18 @@ mux {
 				groups = "uart_cts_b0",
 				       "uart_rts_b0";
 				function = "uart_b";
 				bias-disable;
 			};
+			
+	    xtal_32k_out_pins: xtal-32k-out {
+			mux {
+				groups = "xtal_32k_out";
+				function = "xtal";
+				bias-disable;
+			};
+		};	
 		};
 	};
 };
 
 &ahb_sram {
-- 
Created with Armbian build tools https://github.com/armbian/build


